### PR TITLE
Remove aggregation interface

### PR DESCRIPTION
**Pull request by bbenligiray**
Addresses API3-02

Makes `aggregateBeacons()` private. As a note, the response plan mentioned having `updateBeaconSetWithBeacons()` return the updated Beacon set to replace the `aggregateBeacons()` interface in function. However, this doesn't really work because `updateBeaconSetWithBeacons()` reverts if it can't update the Beacon set. The reader forcing Beacon set updates beforehand is not the intended user flow so I decided against reworking the interface just to enable this. This is still possible in a few ways:
- The reader `tryMulticall()`s a `updateBeaconSetWithBeacons()` and the read call
- The reader calls `updateBeaconSetWithBeacons()` in a try-catch statement
- The reader makes a low-level call to `updateBeaconSetWithBeacons()` and doesn't care if the call fails

### CODE REVIEW

File: contracts/dapis/DapiServer.sol [hunk 1]
```diff
@@ -996,6 +970,32 @@ contract DapiServer is
...
+    /// @notice Called privately to aggregate the Beacons and return the result
+    /// @dev Tha aggregation of Beacons may have a different value than the
+    /// respective Beacon set, e.g., because the Beacon set has been updated
+    /// using signed data
+    /// @param beaconIds Beacon IDs
+    /// @return value Aggregation value
+    /// @return timestamp Aggregation timestamp
+    function aggregateBeacons(
+        bytes32[] memory beaconIds
+    ) private view returns (int224 value, uint32 timestamp) {
```
Comment (acenolaza) [inline contracts/dapis/DapiServer.sol:982]: I remember a discussion about this contract being sort of a base contract for other use cases of DapiServer in the future. Do you think this function might be useful in the contracts inheriting from this one? If so, should we make this function internal instead?

File: contracts/dapis/DapiServer.sol [hunk 2]
```diff
@@ -996,6 +970,32 @@ contract DapiServer is
         }
     }
 
+    /// @notice Called privately to aggregate the Beacons and return the result
+    /// @dev Tha aggregation of Beacons may have a different value than the
+    /// respective Beacon set, e.g., because the Beacon set has been updated
+    /// using signed data
```
Comment (acenolaza) [inline contracts/dapis/DapiServer.sol:976]: Can beacon sets still be updated using signed data?

**Review by acenolaza — APPROVED** [review]
LGTM 👍🏻 (just questions to double check things)

File: contracts/dapis/DapiServer.sol (hunk 2 above)
Comment (acenolaza) [reply contracts/dapis/DapiServer.sol:976]: nvm, found [this](https://github.com/api3dao/airnode-protocol-v1/pull/105/files#diff-47e178e1efdc14225ddb1d731a2e39c7b341830804023013586a1893265ef2b6L974-L976)

**Comment by Ashar2shahid** [comment]
I'm looking at API3-02 and don't really see the reasoning behind making this private/internal. I can see this function being useful where dapps run a subgraph on top of DapiServer and call this function to get the most upto date price and then multicall the users call with updating the beaconSet. If no update is possible they would execute the user's call directly. This is just 1 use case

File: contracts/dapis/DapiServer.sol (hunk 1 above)
Comment (bbenligiray) [reply contracts/dapis/DapiServer.sol:982]: It feels like it's more likely that we will either not modify DapiServer or the modifications needed will be significant enough that we will not be able to reuse the current contracts without modification, and the things we do to make them more reusable will do nothing but reduce readability. I will see if these can be addressed in a final PR.

**Comment by bbenligiray** [comment]
> the reasoning behind making this private/internal

This being publicly visible is not a security issue in the first place, but a usability issue. Reading the data feed with the Beacon set ID or calling `aggregateBeacons()` with the Beacon IDs that constitute the same Beacon set returns different values (as it should), even though they may appear to be equivalent (so the concern is that the users will fail to make the distinction).
- I'd prefer ticking off an audit issue to treating a made up, selfish usage scenario as a first class citizen
- The usage scenario is still supported in the ways listed in the description, though the user will have to update the Beacon set, which they should do anyway

**Comment by bbenligiray** [comment]
(I now get what @Ashar2shahid means, this had nothing to do with the issue in the report)

**Pull request merged**